interceptors/logging/interceptors.go

```diff
@@ -22,65 +22,132 @@ func extractFields(tags tags.Tags) Fields {
 }
 
 type reporter struct {
-	ctx             context.Context
-	typ             interceptors.GRPCType
-	service, method string
-
-	opts   *options
-	logger Logger
+	ctx                        context.Context
+	typ                        interceptors.GRPCType
+	service, method            string
+	firstRequestMessageLogged  bool
+	firstResponseMessageLogged bool
+	isServer                   bool
+	opts                       *options
+	logger                     Logger
 
 	kind string
 }
 
+func (c *reporter) addMetaDetails(logger Logger) Logger {
+	if c.isServer {
+		logger = logger.With("component", "server")
+	}
+	if !c.isServer {
+		logger = logger.With("component", "client")
+	}
+	// Attach the kind, protocol and type as a structured log entry.
+	logger = logger.With("kind", c.kind)
+	logger = logger.With("type", string(c.typ))
+	logger = logger.With("protocol", "grpc")
+
+	return logger
+}
+
+// PostCall logs the server/method name details and error if any.
 func (c *reporter) PostCall(err error, duration time.Duration) {
 	if !c.opts.shouldLog(interceptors.FullMethod(c.service, c.method), err) {
 		return
 	}
 	if err == io.EOF {
 		err = nil
 	}
+	logger := c.addMetaDetails(c.logger)
```

Comment:
Some of the things we can share with other calls instead of adding metas all the time.